### ISSUE TITLE
Make obscured password deterministic

### ISSUE THREAD

**Issue by vnghia** [rclone/rclone#7393]
#### What problem are you are trying to solve?
Obscuring password might not be desirable for some people, see #6423. Its randomness makes it even harder to control its side effect, especially in the infra-as-code environment where the configuration file is generated and managed by a third-party tools such as Terraform or Pulumi.

#### How do you think rclone should be changed to solve that?
We should either:
- Add an option to set the seed for the `iv` in the AES crypt.
- Or much more simpler, set the `iv` to something constant in the first place, just like what we are doing with the `cryptKey`.

This won't create any security issues since if someone know the obscured password, they can always deobscure to the original password regardless of the `iv`.

https://github.com/rclone/rclone/blob/6092fe2aaafb089e4d2a56d55fedecc68f79aa88/fs/config/obscure/obscure.go#L48-L54

I can send a PR if this FR makes sense. Thank you for a wonderful software :smile: 

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.
/cc @ncw

**Comment by ncw** [comment]
> * Or much simpler, set the `iv` to something constant in the first place, just like what we are doing with the `cryptKey`.

This approach is tempting. This purpose of this is to obscure the passwords, however a constant iv makes `aes.CTR` effectively just XOR with a constant value....

> * Add an option to set the seed for the `iv` in the AES crypt.

I probably prefer this option. How about just making it a flag `--obscure-constant` which introduces a constant IV (could even be all nulls).

**Comment by vnghia** [comment]
How about making `--obscure-constant` a boolean option which will:
- either use a random IV.
- or use an IV that is predefined in our code since the IV requires exactly 16 bytes which is a little bit annoying if we force our user enter it.

**Comment by ncw** [comment]
Yes, sounds good. (That is what I meant by flag but I realise that I didn't make myself very clear!)

**Comment by jleeuwes** [comment]
Why not make a constant IV the default instead of adding a flag? Since the purpose of obscuring passwords, if I understood correctly, is only to protect against shoulder surfing, this does not weaken security. It would also be backwards compatible with existing passwords obscured with a random IV, so no flag is needed for that as well.

**Comment by vnghia** [comment]
Hi @ncw, any update about this one ? I also prefer to set the iv to a constant value :/
